src/core/lib.lua

```diff
@@ -715,6 +706,43 @@ function getenv (name)
    else return nil end
 end
 
+-- Return an array of random bytes.
+function random_bytes_from_dev_urandom (count)
+   local bytes = ffi.new(ffi.typeof('uint8_t[$]', count))
+   local f = syscall.open('/dev/urandom', 'rdonly')
+   local written = 0
+   while written < count do
+      written = written + assert(f:read(bytes, count-written))
+   end
+   return bytes
+end
+
+function random_bytes_from_math_random (count)
+   local bytes = ffi.new(ffi.typeof('uint8_t[$]', count))
+   for i = 0,count-1 do bytes[i] = math.random(0, 255) end
+   return bytes
+end
+
+function randomseed (seed)
+   seed = tonumber(seed)
+   if seed then
+      local msg = 'Using deterministic random numbers, SNABB_RANDOM_SEED=%d.\n'
+      io.stderr:write(msg:format(seed))
+      math.randomseed(seed)
+      -- When setting a seed, use deterministic random bytes.
+      random_bytes = random_bytes_from_math_random
+   else
+      -- Otherwise use /dev/urandom.
+      seed = ffi.cast('uint32_t*', random_bytes_from_dev_urandom(4))[0]
+      random_bytes = random_bytes_from_dev_urandom
```

Comment:
Apparently `random_bytes` only exists if `randomseed` is called. It seems this is something the user should know beforehand before using `random_bytes`. Wouldn't be better that `randombytes` always exists and takes a default (possibly `random_bytes_from_dev_urandom`)?